web/src/test/java/org/cbioportal/web/util/ClinicalDataBinUtilTest.java

```diff
@@ -0,0 +1,791 @@
+        keys.add("genie_bpc_testGENIE-MSK-P-0009680");
+        keys.add("genie_bpc_testGENIE-MSK-P-0012393");
+        keys.add("genie_bpc_testGENIE-MSK-P-0017284");
+        
+        return keys;
+    }
+    
+    private List<String> mockFilteredStudyPatientUniqueKeys() {
+        List<String> keys = new ArrayList<>();
+        
+        keys.add("genie_bpc_testGENIE-MSK-P-0015492");
+        keys.add("genie_bpc_testGENIE-MSK-P-0009680");
+        
+        return keys;
+    }
+    
+    private List<String> mockUnfilteredSampleIds() {
+        List<String> sampleIds = new ArrayList<>();
+
+        sampleIds.add("GENIE-MSK-P-0003156-T01-IM5");
+        sampleIds.add("GENIE-MSK-P-0009680-T01-IM5");
+        sampleIds.add("GENIE-MSK-P-0012393-T01-IM5");
+        sampleIds.add("GENIE-MSK-P-0012393-T02-IM6");
+        sampleIds.add("GENIE-MSK-P-0015492-T01-IM6");
+        sampleIds.add("GENIE-MSK-P-0017284-T01-IM6");
+        sampleIds.add("GENIE-MSK-P-0017284-T02-IM5");
+
+        return sampleIds;
+    }
+    
+    private List<String> mockFilteredSampleIds() {
+        List<String> sampleIds = new ArrayList<>();
+
+        sampleIds.add("GENIE-MSK-P-0009680-T01-IM5");
+        sampleIds.add("GENIE-MSK-P-0015492-T01-IM6");
+
+        return sampleIds;
+    }
+
+    private List<String> mockUnfilteredPatientIds() {
+        List<String> patientIds = new ArrayList<>();
+
+        patientIds.add("GENIE-MSK-P-0015492");
+        patientIds.add("GENIE-MSK-P-0003156");
+        patientIds.add("GENIE-MSK-P-0009680");
+        patientIds.add("GENIE-MSK-P-0012393");
+        patientIds.add("GENIE-MSK-P-0017284");
+
+        return patientIds;
+    }
+    
+    private List<String> mockFilteredPatientIds() {
+        List<String> patientIds = new ArrayList<>();
+
+        patientIds.add("GENIE-MSK-P-0015492");
+        patientIds.add("GENIE-MSK-P-0009680");
+
+        return patientIds;
+    }
+    
+    private List<Patient> mockUnfilteredPatients() {
+        return mockPatients(mockUnfilteredPatientIds());
+    }
+
+    private List<Patient> mockFilteredPatients() {
+        return mockPatients(mockFilteredPatientIds());
+    }
+    
+    private List<Patient> mockPatients(List<String> patientIds) {
+        List<Patient> patients = new ArrayList<>();
+        
+        for (String patientId: patientIds) {
+            Patient patient = new Patient();
+            patient.setCancerStudyIdentifier(STUDY_ID);
+            patient.setStableId(patientId);
+            patients.add(patient);
+        }
+        
+        return patients;
+    }
+
+    private List<String> mockUnfilteredStudyIds() {
+        List<String> studyIds = new ArrayList<>();
+
+        studyIds.add(STUDY_ID);
+        studyIds.add(STUDY_ID);
+        studyIds.add(STUDY_ID);
+        studyIds.add(STUDY_ID);
+        studyIds.add(STUDY_ID);
+        studyIds.add(STUDY_ID);
+        studyIds.add(STUDY_ID);
+
+        return studyIds;
```

Comment:
Since we have a lot this kind of code, maybe we should create a function `populateStudyIds`?

Reply:
Good point. Seems like there is already a utility method called `nCopies`. So something like this should work:

```Java
return Collections.nCopies(7, STUDY_ID);
```